File: common/src/main/java/org/flanigan/proxyhook/common/QuietError.java
package org.flanigan.proxyhook.common;

import java.io.PrintStream;
import java.io.PrintWriter;

// UGLY HACK
class QuietError extends Error {

    @Override
    public synchronized Throwable fillInStackTrace() {
        return this;
    }

    @Override
    public StackTraceElement[] getStackTrace() {
        return new StackTraceElement[0];
    }

    @Override
    public void printStackTrace() {
    }

    @Override
    public void printStackTrace(PrintStream s) {
    }

    @Override
    public void printStackTrace(PrintWriter s) {
    }
}

File: common/src/main/java/org/flanigan/proxyhook/common/QuietError.kt
package org.flanigan.proxyhook.common

import java.io.PrintStream
import java.io.PrintWriter

/**
 * @author Sean Flanigan [sflaniga@redhat.com](mailto:sflaniga@redhat.com)
 */
// UGLY HACK
internal class QuietError : Error() {

    override fun fillInStackTrace(): Throwable {
        return this
    }

    override fun getStackTrace(): Array<StackTraceElement> {
        return emptyArray()
    }

    override fun printStackTrace() {}

    override fun printStackTrace(s: PrintStream) {}

    override fun printStackTrace(s: PrintWriter) {}
}
